### ISSUE TITLE
NVDA does not navigate the text in edit field while browsing in google chrome

### ISSUE THREAD

**Issue by abdullah5490** [nvaccess/nvda#6699]
Hi NVDA developers and contributers. First of all i would like to extremely congratulate you on bringing such a nice screen reading solution for windows OS, even for free! I really appreciate this project, and using NVDA as the default screen reader. Unable to donate yet being a student, but my moral support is always with you.
Now, let's come to the point. As the title said, I've tried a lot to use chrome as the default browser, but an issue preventing me to do so. When i enter text in an edit field on any web page, NVDA doesn't navigate that entered text in the field properly. For example, if i've typed my e-mail address on the gmail login screen which is abdullah5490@gmail.com, when i try to read the insurted text by characters with left/right aerrow, it will not read the text in proper seequence. In this case, i press home key to move curser at the start of the text, on the character "a", now when i press right aerrow, it will speak the next character which is "b", when i press right aerrow again to read the next character, it will repeat the previous character which was "a", or the current character which is "b". even the curser has moved to the next character "d", but NVDA ocasionally speak it at that time, but mostly, it went back speaking the previously crossed characters. Similarly, if i want to check the previous character by left aerrow, it will speak even the next character instead of previous one, or sometime it said "blank", even there is a text in the field.
I hope i am able to clarify my point in front of you. Due to this problem, it is impossible to read the insurted text in a field properly, and duet o improper seequence while navigating it, The user will unable to understand what they have typed in the field. This issue began in the latest release of 2016.4. I am using chrome latest version 55.0.2883.87, latest at the time of this post. 
Looking for your possitive comments and feebback.

**Comment by jcsteh** [comment]
I've seen reports of something like this issue before, but not with recent
versions of Chrome. In the example you gave, when you pressed right arrow
from "b", you said it sometimes says "a". This I've never seen/heard of and
I'm not sure how it's possible. Are you absolutely certain of this; can you
test again to be sure? I can see how it might say "b" again instead of "d",
but "a" suggests a very different issue I can't even begin to fathom. The
important point is that I can understand why the character you were just on
might be repeated, but I can't see any way the character before could be
reported. Thanks.

**Comment by jcsteh** [comment]
Also, is it possible that Chrome was updated on your system recently as
well? I can't think of any change in 2016.4 that should have affected this
in any way.

**Comment by abdullah5490** [comment]
No, i've tried a lot on different web pages to check it out before 
writing the above issue to you. Even, before posting here, i've tried to 
write this article on chrome, but even failed to read out my own writing 
there. Then i switches to firefox to post it here. And this issue was 
not present there before this release, and i've mentioned the build 
number of chrome in my text as well, which is the currently latest 
version. And this version is not so recent 0 it was also release about a 
month ago. And everything written in my post is the detailed experience 
of my testing.

Yes i'm also wondering why is it happening, but the situation is, that 
it is actually happening! I've never installed any add-on to chrome, and 
also, no add-on to NVDA except eloquence and NVDA remote, which, 
definetly, does not have any relation with chrome's accessibility.

Using win10x64 with the chrome 64bit edition.


regards,
Abdullah Zafar from Islamabad Pakistan!
CEO at
Blind Help Project <http://www.blindhelp.net>
Trainning coordinator at
Masiha Educational Welfare Society <http://www.facebook.com/masews>
Media manager at
Pakistan Paraclimbing Club <http://www.facebook.com/pakparaclimbing>
Add me on skype as Abdullah.93@outlook.com
Meet on facebook
Abdullah Zafar on facebook! <http://www.facebook.com/abdullah5490>
Or on twitter
Abdullah5490 on twitter! <http://www.twitter.com/abdullah5490>
Send from thunderbird for windows e-mail client!

**Comment by jcsteh** [comment]
Okay. I can sometimes reproduce the current character being repeated here.
I can't ever reproduce the character before the current one being reported,
which is going to make this difficult to investigate.

**Comment by abdullah5490** [comment]
You can even connect to me personally through remote access and 
investigate the issue yourself

regards,
Abdullah Zafar from Islamabad Pakistan!
CEO at
Blind Help Project <http://www.blindhelp.net>
Trainning coordinator at
Masiha Educational Welfare Society <http://www.facebook.com/masews>
Media manager at
Pakistan Paraclimbing Club <http://www.facebook.com/pakparaclimbing>
Add me on skype as Abdullah.93@outlook.com
Meet on facebook
Abdullah Zafar on facebook! <http://www.facebook.com/abdullah5490>
Or on twitter
Abdullah5490 on twitter! <http://www.twitter.com/abdullah5490>
Send from thunderbird for windows e-mail client!

**Comment by jcsteh** [comment]
That may be an option if fixing the first issue doesn't resolve the second.
Thanks for offering.

**Comment by feerrenrut** [comment]
This issue seems to be quite related to #6424. I will mark this issue as blocked until #6424 is resolved then we can see if this issue is also resolved.

@abdullah5490 Could you provide a log file (at debug level) for this issue. You will need to set the logging level first, then reproduce the issue, and then you can attach the log to this issue. This will help us to reproduce the issue or at least gain some further insight. It will be unlikely that we can resolve this issue without some further information. For detailed information on how to provide these logs please see [the wiki page for log files and crash dumps](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps)

**Comment by abdullah5490** [comment]
Sure, why not, i'll definetly provide the logs, and will help you from 
dawn to death to make a bug-free NVDA for all!

regards,
Abdullah Zafar from Islamabad Pakistan!
CEO at
Blind Help Project <http://www.blindhelp.net>
Trainning coordinator at
Masiha Educational Welfare Society <http://www.facebook.com/masews>
Media manager at
Pakistan Paraclimbing Club <http://www.facebook.com/pakparaclimbing>
Add me on skype as Abdullah.93@outlook.com
Meet on facebook
Abdullah Zafar on facebook! <http://www.facebook.com/abdullah5490>
Or on twitter
Abdullah5490 on twitter! <http://www.twitter.com/abdullah5490>
Send from thunderbird for windows e-mail client!

**Comment by abdullah5490** [comment]
Hi Mr. @feerrenrut and @jcsteh 
Here is the log at the time of problem. here what i'am done for creating the log that you will find in the log itself, but i am stating here for your ease in understanding.
1. I change the logging level to "debug" by going in the general settings. 
2. afterward, i launched google chrome from my desktop. the curser focussed at the address bar.
3. then i typed "facebook.com" to open the facebook page.
4. Then i tried to find an edit field for writing a comment, by using the navigation quick key command "e" and "shift+e".
5. After reaching to the "write a comment" field, i pressed enter to go to the editting mode.
6. Now i typed the text "i am testing NVDA in chrome while writing in the edit field and then i will read it with the use of aerrows/home/end keys. now i am going to read it, please note the problem here!".
7. Now i wanted to read the written text, here's what happened:
- Upon pressing left aerrow continuously to go backward in the text, i can't understand what's written due to completely unknown seequence of reading.
- Then tried to read in the forward direction by right aerrow, but again unable to understand, what's written, where my curser is hunting, and which word/character is currently be focussed actually, due to totally strange reading order.
- Then i tried to read word-by-word with control+right/left aerrows. This time it is little bit understandable to some extent, because NVDA is speaking the whole words. But the problem of reading order still remain same. NVDA does not read in sequence. It sometime skip the words, and sometime announcing the words twice or thrice.
8. Then i closed the chrome windows with control+w, open the log viewer from NVDA menu, saved  it to the file, and attached it below.
[nvda.txt](https://github.com/nvaccess/nvda/files/689800/nvda.txt)

**Comment by jcsteh** [comment]
I believe this to be a duplicate of #6424, which is fixed by #7201 (now in master). Closing as duplicate. Please reopen if this problem persists. Thank you.

**Comment by abdullah5490** [comment]
I can't find the milestone download. Can you please paste a link to download the milestone release? Please? I'll test it and report you back about the issue.

**Comment by abdullah5490** [comment]
well, here is my report. in the latest commit/master release, the NVDA is now navigating absolutely fine as character by character. but when I try to read word by word in an edit field under chrome, it read out all the content in that field, not each word one by one. For example, after typing this comment in the edit field using chrome, I pressed ctrl+right/left, it read my whole text written in this edit field. Not each word. This is only the case of chrome. in firefox, it just work fine.

**Comment by jcsteh** [comment]
This sounds like a different issue. Does this happen in all edit fields in Chrome or just some? Do you have Chrome set to use a language other than English, and if so, what language?

**Comment by abdullah5490** [comment]
@jcsteh No, i'm using the english version of chrome, because It is the only language that I understand internationally. and its w10x64
